haystack/nodes/retriever/multimodal/retriever.py

```diff
@@ -205,18 +205,15 @@ def retrieve_batch(  # type: ignore
         query_embeddings = self.query_embedder.embed(documents=query_docs, batch_size=batch_size)
 
         # Query documents by embedding (the actual retrieval step)
-        documents = []
-        for query_embedding, query_filters in zip(query_embeddings, filters_list):
-            docs = document_store.query_by_embedding(
-                query_emb=query_embedding,
-                top_k=top_k,
-                filters=query_filters,
-                index=index,
-                headers=headers,
-                scale_score=scale_score,
-            )
+        documents = document_store.query_by_embedding_batch(
+            query_embs=query_embeddings,  # type: ignore
```

Comment:
Would this stop MultiModal retriever from using all the document stores?

Reply:
My understanding is that any document store still works because the default implementation is just to run `query_by_embedding` and only `OpenSearchDocumentStore` and `ElasticsearchDocumentStore` have a special implementation. Our tests use the MultiModalRetriever only with `InMemoryDocumentStore` if I am not mistaken. So we should double check that manually before merging. @tstadel You could try to run the tests involving MultiModalRetriever in test_retriever.py with ElasticsearchDocumentStore, maybe?

Reply:
@julian-risch @mayankjobanputra Ok I tested it all multimodal tests of `test_retriever.py` locally and added another one that uses `retrieve_batch`. All tests were fine. I added the `retrieve_batch` test using `InMemoryDocumentStore` as all the others.
I know that `#type: ignore` is not really nice here, so I'll open another PR to resolve this. It would be just too big for this one I guess.